# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.h
# Commit: a45df1876e37
# Full Hash: a45df1876e375459648251484c7d5552d2bdae6d
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-01 16:21:08
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r 1a85cc92d2fb -r a45df1876e37 netwerk/cookie/CookieCommons.h
--- a/netwerk/cookie/CookieCommons.h	Mon Jun 01 10:29:06 2020 +0000
+++ b/netwerk/cookie/CookieCommons.h	Mon Jun 01 10:29:34 2020 +0000
@@ -108,6 +108,15 @@
       nsIChannel* aChannel);
 
   static bool ShouldIncludeCrossSiteCookieForDocument(Cookie* aCookie);
+
+  static bool MaybeCompareScheme(Cookie* aCookie,
+                                 nsICookie::schemeType aSchemeType);
+
+  static nsICookie::schemeType URIToSchemeType(nsIURI* aURI);
+
+  static nsICookie::schemeType PrincipalToSchemeType(nsIPrincipal* aPrincipal);
+
+  static nsICookie::schemeType SchemeToSchemeType(const nsACString& aScheme);
 };
 
 }  // namespace net